lib/facility.rb

```diff
@@ -1,14 +1,67 @@
+require 'date'
+
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_accessor :name, :address, :phone, :services, :registered_vehicles, :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
-    @services = []
+  def initialize(facility_details)
+      @name = facility_details[:name]
+      @address = facility_details[:address]
+      @phone = facility_details[:phone]
+      @services = []
+      @registered_vehicles = []
+      @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    @vehicle = vehicle
+    @facility = self
+
+    if @facility.services.include?('Vehicle Registration')
+      @vehicle.registration_date = Date.today
+      @facility.registered_vehicles << vehicle
+
+      if @vehicle.antique?
+        @vehicle.plate_type = :antique
+        @facility.collected_fees += 25
+      elsif @vehicle.electric_vehicle?
+        @vehicle.plate_type = :ev
+        @facility.collected_fees += 200
+      else
+        @vehicle.plate_type = :regular
+        @facility.collected_fees += 100
+      end
+    end
+  end 
```

Comment:
Referring to lines 20 and 21, I see you doing this a lot in your methods -- reassigning values to be instance variables. Unless those values need to be called outside of that method, there's no need to make them instance variables. You could redo this entire method to look something like: 

```ruby 
def register_vehicle(vehicle)
    if @services.include?('Vehicle Registration')
      vehicle.registration_date = Date.today
      @registered_vehicles << vehicle

      if vehicle.antique?
        vehicle.plate_type = :antique
        @collected_fees += 25
      elsif vehicle.electric_vehicle?
        vehicle.plate_type = :ev
        @collected_fees += 200
      else
        vehicle.plate_type = :regular
        @collected_fees += 100
      end
    end
  end 
```

Reply:
In your submission, you asked about how you could avoid nested if statements in this method. It is totally okay to have nested if's in a method, but something you could do is create a helper method that would handle the functionality for assigning plate_type and collecting fees.

```ruby
def register_vehicle(vehicle)
    if @services.include?('Vehicle Registration')
      vehicle.registration_date = Date.today
      @registered_vehicles << vehicle
       assign_plate_type(vehicle)
       collect_fees(vehicle)
    end
  end 

def assign_plate_type(vehicle)
  if vehicle.antique?
        vehicle.plate_type = :antique
      elsif vehicle.electric_vehicle?
        vehicle.plate_type = :ev
      else
        vehicle.plate_type = :regular
      end
end 

def collect_fees(vehicle)
  if vehicle.antique?
        @collected_fees += 25
      elsif vehicle.electric_vehicle?
        @collected_fees += 200
      else
        @collected_fees += 100
      end
end
```